### ISSUE TITLE
Best way to pass keys to modules?

### ISSUE THREAD

**Issue by Logon27** [patrick-kidger/equinox#1083]
I was looking at the flax docs and they have...
https://flax-linen.readthedocs.io/en/latest/guides/flax_fundamentals/rng_guide.html#receiving-manipulating-and-creating-prng-keys-with-module-make-rng
There also seems to be a related stackoverflow post that was removed for some reason...
https://stackoverflow.com/questions/79608825/correct-way-of-providing-prngkeyarray-to-dropout-layer-using-jax-and-equinox

I was just wondering if there was any functionally equivalent way of handling the keys in a clean way that is not so painful? Mainly because I have some dropout layers but want a cleaner way to manage the keys required for them.

**Comment by patrick-kidger** [comment]
Equinox avoids using stateful PRNGs in favour of passing them around explicitly. This is in-line with the underlying JAX style, and reflects my preference for maintaining tighter control over the randomness, e.g. to obtain reproducibility from run-to-run. Thus `jax.random.split`ing your keys each time you need them is the recommended approach.

**Comment by Logon27** [comment]
Fair enough, was just curious. Afaik that rng stream is reproduceable its just a convenience.
